moveit_planners/pilz_industrial_motion_planner/test/unittest_trajectory_generator_circ.cpp

```diff
@@ -699,6 +709,58 @@ TEST_P(TrajectoryGeneratorCIRCTest, CenterPointPoseGoalFrameIdBothConstraints)
   checkCircResult(req, res);
 }
 
+/**
+ * @brief Set different frame_id's on both goal pose and path constraint
+ */
+TEST_P(TrajectoryGeneratorCIRCTest, CenterPointPoseGoalFrameIdPoseAndPathConstraints)
+{
+  auto circ{ tdp_->getCircCartCenterCart("circ1_center_2") };
+
+  moveit_msgs::MotionPlanRequest req = circ.toRequest();
+
+  // set to start state
+  auto rstate = planning_scene_->getCurrentState();
+  moveit::core::robotStateMsgToRobotState(planning_scene_->getTransforms(), req.start_state, rstate);
+  rstate.update();
+
+  // compute offset between links
+  auto pose_tcp = rstate.getFrameTransform("prbt_tcp");
+  auto pose_link5 = rstate.getFrameTransform("prbt_link_5");
+
+  double offset_link5_to_tcp = (pose_link5.inverse() * pose_tcp).translation()[1];
```

Comment:
Why do you constrain the offset to the y-component?

Reply:
Because it is align with the z axis tcp link, which the goal pose is derived. To be able to place the center circle at the center of the goal and start state. I think the image below will is a little more descriptive.